micrometer-core/build.gradle

```diff
@@ -239,6 +239,17 @@ tasks.register("shenandoahTest", Test) {
     jvmArgs '-XX:+UseShenandoahGC'
 }
 
+tasks.register("generationalShenandoahTest", Test) {
+    // set heap size for the test JVM(s)
+    maxHeapSize = "1500m"
+
+    useJUnitPlatform {
+        includeTags 'gc'
+    }
+
+    jvmArgs '-XX:+UseShenandoahGC', '-XX:+UnlockExperimentalVMOptions', '-XX:ShenandoahGCMode=generational'
```

Comment:
We can remove `-XX:+UnlockExperimentalVMOptions` and run this test task only on our Java 25+ builds when we have any. In the meantime, we can run it on Java 24 with this option.